pkg/target_determinator.go

```diff
@@ -730,9 +731,46 @@ func runToCqueryResult(context *Context, pattern string) (*analysis.CqueryResult
 	if err := proto.Unmarshal(content, &result); err != nil {
 		return nil, fmt.Errorf("failed to unmarshal cquery stdout: %w", err)
 	}
+
+	compatibleTargets, err := findCompatibleTargets(context, pattern)
+	if err != nil {
+		return nil, err
+	}
+
+	// Filter out incompatible targets.
```

Comment:
Filtering the targets here also filters transitive dependencies, which I am not sure is needed. If transitive dependencies do not need to be filtered, then filtering can happen in doQueryDeps while populating MatchingTargets. WDYT?

Reply:
Generally this PR LGTM (including the gofmt comment formatting changes), but yeah, let's move this to filter `MatchingTargets` rather than all transitive deps

Also, can you add a new boolean field to `Context`: `FilterIncompatibleTargets` to control whether we do this filtering? Some TD users try to do multi-platform analysis (though I'm not sure how accurately, given `cquery` will change behaviour based on platform) using this logic, so preserving the escape hatch for them to be able to do so would be useful. You can wire up in cli/flags.go to set a value based on a flag (which I think we should default to `true`).

Lastly, can you add a test for this behaviour? I guess just adding a target which is incompatible with all platforms would be sufficient. I just put together https://github.com/bazel-contrib/target-determinator/pull/48 with instructions for how to add tests, because it's slightly non-trivial - I think in this instance, this is a "general" test that should apply to all TD implementations, so should live in Tests.java :)